### PR TITLE
Fix promise callback error handling

### PR DESCRIPTION
**Pull request by danlamanna**
While working on #48 I found various errors that weren't being visibly thrown inside of the example notebooks which led to a mismatch in state between the client and server. In our promise implementation we need to catch any exceptions that get thrown in the callbacks of our original promise.

This fixes the issues with the notebooks and in the future exceptions will actually be propagated to the logger. To test this just run notebook 2 or 3 from 6a5e152 (before I fixed them) to watch the exceptions being logged.

The tl;dr:
```
# Won't catch errors thrown in some_callback, everything will appear to be fine
remote.set_center(x, y, z).then(some_callback)

# some_handler will be notified of errors that occur in some_callback
remote.set_center(x, y, z).then(some_callback).catch(some_handler)
```

### CODE REVIEW

File: geonotebook/kernel.py [hunk 1]
```diff
@@ -365,17 +371,17 @@ def _add_layer(layer_name):
             params['zIndex'] = len(self.layers)
 
             cb = self._remote.add_wms_layer(layer.name, layer.vis_url, params)\
-                .then(_add_layer, self.rpc_error)
```
Comment (aashish24) [inline geonotebook/kernel.py]: @danlamanna wouldn't it be possible to have a method that can avoid this code duplication of attaching error handler specially for add layer calls?

Comment (danlamanna) [inline geonotebook/kernel.py]: Are you talking about the 3 calls to `add_*_layer` functions? If so, #48 refactors this method so it only calls `self._remote.add_layer` once with the type as an argument, see [here](https://github.com/OpenGeoscience/geonotebook/blob/c63f446eed0884eb75b5e40706a9af54b5fba293/geonotebook/kernel.py#L389).

Comment (jbeezley) [inline geonotebook/kernel.py]: You could just `return cb.catch(self.callback_error)` below.

**Comment by aashish24** [comment]
> Are you talking about the 3 calls to add_*_layer functions? If so, #48 refactors this method so it only calls self._remote.add_layer once with the type as an argument, see here.

Ah, yes, thanks.

**Comment by aashish24** [comment]
> You could just return cb.catch(self.callback_error) below.

+1

File: geonotebook/kernel.py (hunk 1 above)
Comment (jbeezley) [inline geonotebook/kernel.py]: This is going away soon anyway, so we can probably just leave it as is.
